### PR TITLE
11 tables frames

### PR DESCRIPTION
**Pull request by nicusoara1**

### CODE REVIEW

**Comment by StefanoSperetta** [comment]
Looks good! 
Any reason _ not allowed in the password regex (https://stackoverflow.com/questions/3721843/how-to-allow-underscore-in-this-password-complexity-regex)? 
Do you think the regex would survive a SQL injection test? It seems (but I am a little rusty with regexps, that a blank might pass the check.

**Comment by iiacoban42** [comment]
Indeed _ not being allowed is a bug! This regex from stack overflow seems to work for _ `(?=^.{8,}$)(?=.*[a-z])(?=.*[A-Z])(?=.*[\W_])(?=^.*[^\s].*$).*$` or we could split it to check separately for each password condition: length, 1 uppercase and lowercase letter, number and special character. @chaiwon59 please have a look at this before you make your PR for the login.

With regard to the SQL injection, as long as we use parameterized queries and the object-relational mapping from Django, it should be secure. Raw SQL queries are possible though, but come with the SQL injection risks, I would avoid those as much as possible indeed. https://docs.djangoproject.com/en/3.2/topics/security/. So, in our setup if a user decides to have an SQL query as password, it shouldn't cause any problems. For fun I changed the password of one of my social media accounts to `Select * from Users LIMIT 100;` then changed it back with no issues :). @StefanoSperetta do you see any potential risks associated with this?

image

File: src/ewilgs/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Generated by Django 3.2.8 on 2021-11-14 11:42
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    initial = True
```
Comment (iiacoban42) [inline src/ewilgs/migrations/0001_initial.py:9]: Can you please add the migration folders to the gitignore? For this to take effect you also need to delete them and commit the changes.

**Comment by StefanoSperetta** [comment]
Can you try with a username:

- admin' or 1=1 -- 
- admin" or 1=1 --

(the idea comes from here: https://portswigger.net/support/using-sql-injection-to-bypass-authentication). These 2 usernames should force the authentication of the admin user even without a password (just to be paranoid): I guess the ' and " should be caught and the blanks should be flagged as well so I guess we should be fine.

**Comment by iiacoban42** [comment]
image
Doesn't allow a blank password field, tried both ' and ". This is for Django admin, @chaiwon59 will make another PR will our own login.

**Pull request merged**